src/presets.asm

```diff
@@ -198,58 +182,119 @@ preset_load_preset:
     RTL
 }
 
-preset_to_memory:
+category_preset_load:
```

Comment:
There's a lot in this method.  I still need to run more tests.  I have verified that crossbank works, but there are three possible scenarios for when the crossbank happens and I'm not sure I've tested all three yet.

I did simplify the preset data format.  Previously it was dl $7Exxxx, db $02, dw $xxxx.  It was always $7E, and $7E is the only value that makes sense.  There was support for db $01 but we never used it.  By reducing the format to dw $xxxx, $xxxx, of course this saves a lot of space, but also it makes everything in the preset data a word.  That makes crossbank support more practical to implement.

I did make an assumption:  Once we have crossed the bank once, we won't again.  This relies on the preset data being less than one bank (we aren't even close to that limit) and it relies on the preset data being laid out in a sane order in the asm file.  If using the lua script I'd expect a sane order, but if doing some work by hand I guess it could be an issue?  I'd prefer the preset data be in order anyway.  It's okay to have gaps (e.g. hi jump preset's parent is kraid instead of bootless preset), but it needs to be in order.

Reply:
Are you aware of how GT Classic is setup with both boots and bootless routes together? Is that the issue you mention here? I believe I did the same thing for LN presets in PRKD. There's a separate group for No-X presets up to Ridley.

Reply:
Yep that's what I was referring to.  It seems like they still work, and I do see the preset data is laid out in a good order so it should work.  A bit of a scare seeing ice/spazer not selected before ridley in the No-X presets... but then I went back to 2.4.0 and same thing so I think there's just an issue with the preset data.

Reply:
I was going to ask, do you have any more category presets, or are they all uploaded at this point?

Reply:
Interesting. I didn't realize that mistake ever made it into the main branch. I planned to remake most of the presets again when the MB preset stuff is finalized, but that's pretty easy to fix anytime.

Reply:
I have KPDR 22% (adds Kraid tank) and PRKD 19% (skips wave missile). There's also a couple NewGame+ routes that I never planned to submit. 

Feel free to delete the All Bosses routes. I made those in a hurry for a multicategory tournament that featured the category and no one ever ran it outside of those races, much less used the presets. They're low quality and mostly wasted space.

Reply:
I think one of them isn't even the correct route. Maybe some lua mistakes that caused a few broken presets. It's been a while and I've had zero motivation to look at them again.